kernel/src/fs/file_handle.rs

```diff
@@ -116,6 +117,20 @@ pub trait FileLike: Pollable + Send + Sync + Any {
     fn as_socket(&self) -> Option<&dyn Socket> {
         None
     }
+
+    fn read_proc_fdinfo(&self, offset: usize, writer: &mut VmWriter) -> Result<usize> {
+        // TODO: Add implementations for specific file types.
+        warn!("The default implementation of read_proc_fdinfo is used, which may be incorrect for some file types.");
+
+        let mut printer = VmPrinter::new_skip(writer, offset);
+
+        writeln!(printer, "pos:\t{}", 0)?;
+        writeln!(printer, "flags:\t0{:o}", self.status_flags().bits())?;
```

Comment:
Actually this is incorrect because (see [the manual](https://man7.org/linux/man-pages/man5/proc_pid_fdinfo.5.html)):
> If the close-on-exec
> file descriptor flag is set, then flags will also include
> the value O_CLOEXEC.

Note that `O_CLOEXEC` is a per-file-**descriptor** (instead of per-file-**description**) flag. So it is not contained in `FileLike`. It is contained in `FileTableEntry` instead. So the design of this API needs to take this into account.